### DISCUSSION TITLE
Отправка файла по почте без сохранения в БД

### DISCUSSION THREAD

**Discussion by Vadimgrig2012** [totumonline/totum-ru-issues-and-discussions#307] [Как это сделать?] [answered]
Добрый день всем. Есть следующая задача: необходимо по нажатию на кнопку отправить письмо выбранному пользователю с шаблоном HTML и файл, но файл необходимо выбрать через системное окно выбора файлов и не сохранять его в БД при отправке и до отправки письма.
Понимаю, как отправить заранее загруженный файл вместе с шаблоном выбранному юзеру, но вот конкретно без загрузки файла в бд - вот тут у меня ступор.

**Comment by erpic-ru** [comment]
Если делать интерфейс отправки на временной таблице, то все загруженное выпилится через 10 минут, вроде так

**Reply by Vadimgrig2012** [reply]
Как вариант. Спасибо!

**Reply by ttm-alexeypolunin** [reply]
Через час. Если удаляется через 10 минут — скорее всего есть рассинхрон времени у PHP, Системы и БД

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Вот так можно вызвать сначала инпут а потом аплоад и аплоад уже отправляет письмо:

```
= : linkToInput(title: "multi_code"; code: $code_1; var: "code_2" = $code_2; button: "Save")

    ```code_1:totum
    =: linkToFileUpload(code: $#code_2; limit: 1; type: "*"; var: "step_1_input" = $#input)
    ```

    ```code_2:totum
    =: emailSend(to: $sel; title: "..."; body: "..."; files: $#input)
    sel: "...трали вали где $#step_1_input равно..."
    ```

```
